utils/env_preferences/tests/test.rs

```diff
@@ -0,0 +1,153 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(target_os = "linux")]
+#[cfg(test)]
+mod linux_tests {
+    use env_preferences::{get_locales, get_system_calendars, LocaleCategory, RetrievalError};
+    use icu_locale::Locale;
+    use libc::setlocale;
+
+    // Testing fetching of locale, as `get_locales` fetches the locales for category
+    // `LC_ALL`. For this category this should return non empty
+    #[test]
+    fn test_get_locales() {
+        let locale_res = get_locales();
+
+        match locale_res {
+            Ok(locale_map) => {
+                assert!(
+                    !locale_map.is_empty(),
+                    "Expected locale_map not to be empty"
+                );
+            }
+            Err(err) => {
+                panic!("{:?}", err)
+            }
+        }
+    }
+
+    #[test]
+    fn test_converting_locales() {
+        let locale_res: std::collections::HashMap<LocaleCategory, String> = get_locales().unwrap();
+        for locale in locale_res.into_values() {
+            let parts: Vec<&str> = locale.split('.').collect();
+
+            // Skipping "C" and those ending with "UTF-8", as they cannot be converted
+            // into the locale
```

Comment:
Why can't they be converted into a `Locale`? Can they be converted into something else? Can we check that the strings are in a form that we expect, even if not a BCP-47 Locale?

Reply:
Tried, doing the following:
```rust
let loc_str = "C";
let locale_ loc.parse.unwrap();
```
It fails

Reply:
Converting to locale is a whole other can of worms, there will be strings like `C`, `de_CH.UTF-8`, etc. Let's not scope-creep this PR and stick with strings for now.

Reply:
I think I've been clear about the line I draw given that this is unsafe code:

1. All code paths must be covered
2. All bytes resulting from calls to unsafe functions must be read
3. The _shape_ of the output should be asserted

I am relying on @robertbastian and @zbraniecki to verify condition 1 (coverage).

I guess condition 2 is covered in `std::ffi::CStr::to_str`.

For condition 3, if the strings are not valid BCP-47, then let's assert what shape they are. The bare minimum would be an assertion that they are ASCII, for example. But condition 3 wouldn't rise to the level of blocking the PR from landing in experimental.